### PR TITLE
built a function to get parameter og a function and appending in vari…

### PR DESCRIPTION
**Pull request by sparsh459**
…able dictionary

### CODE REVIEW

File: vevestaX/vevesta.py [hunk 1]
```diff
@@ -106,6 +106,27 @@ def endModelling(self):
     # create alias of method modellingStart and modellingEnd
     start = startModelling
     end = endModelling
+    
+    
+    # function to get arguments of a function
+    def parameters(self, func):
+        def wrapper(*args, **kwargs):
+            # to get parameters of function that are passed
+            func_args = inspect.signature(func).bind(*args, **kwargs).arguments
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:115]: use camel case and better name

File: vevestaX/vevesta.py [hunk 2]
```diff
@@ -106,6 +106,27 @@ def endModelling(self):
...
+            func_args = dict(func_args)
+            # to get values that are not passed and defautlt values
+            sig = inspect.signature(func)
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:118]: use better name: signature not sig

File: vevestaX/vevesta.py [hunk 3]
```diff
@@ -106,6 +106,27 @@ def endModelling(self):
...
+            for param in sig.parameters.values():
+                if ( param.default is not param.empty):
+                    # checks in ke exist in abouve dictionary then doesn't update it will default value, otherwise append into dictionary
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:121]: please write key not ke

File: vevestaX/vevesta.py [hunk 4]
```diff
@@ -106,6 +106,27 @@ def endModelling(self):
...
+                    if param.name not in func_args:
+                        func_args[param.name] = param.default
+            
+            for key, value in func_args.items():
+                if key not in self.__variables:
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:126]: check if the param value is of primitive data type

**Review by Priyanka-Vevesta — CHANGES_REQUESTED** [review]

File: vevestaX/vevesta.py [hunk 5]
```diff
@@ -106,6 +106,28 @@ def endModelling(self):
...
+            for param in defaultParameters.parameters.values():
+                if ( param.default is not param.empty):
+                    # checks in key exist in abouve dictionary then doesn't update it will default value, otherwise append into dictionary
+                    if param.name not in functionParameters:
+                        functionParameters[param.name] = param.default
+            
+            for key, value in functionParameters.items():
+                if type(value) in [int, float, bool, str]:
+                    if key not in self.__variables:
+                        self.__variables[key] = value
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:128]: write both if stmts in single line

File: vevestaX/vevesta.py [hunk 6]
```diff
@@ -106,6 +106,28 @@ def endModelling(self):
...
+            defaultParameters = inspect.signature(func)
+            for param in defaultParameters.parameters.values():
+                if ( param.default is not param.empty):
+                    # checks in key exist in abouve dictionary then doesn't update it will default value, otherwise append into dictionary
+                    if param.name not in functionParameters:
+                        functionParameters[param.name] = param.default
+            
+            for key, value in functionParameters.items():
+                if type(value) in [int, float, bool, str]:
+                    if key not in self.__variables:
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:127]: see if you can use some python construct to optimize this code...like list comprehension

**Review by Priyanka-Vevesta — CHANGES_REQUESTED** [review]

File: vevestaX/vevesta.py [hunk 7]
```diff
@@ -106,6 +106,29 @@ def endModelling(self):
...
+            # to get values that are not passed and defautlt values
+            defaultParameters = inspect.signature(func)
+            for param in defaultParameters.parameters.values():
+                if ( param.default is not param.empty):
+                    # checks in key exist in abouve dictionary then doesn't update it will default value, otherwise append into dictionary
+                    if param.name not in functionParameters:
+                        functionParameters[param.name] = param.default
+                else:
+                    # this condition checks if the defult value is 0 than variable won't be tracked
+                    if param.default == 0:
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:126]: I dunno why you did this? doesnt seem right

**Review by Priyanka-Vevesta — CHANGES_REQUESTED** [review]

File: vevestaX/vevesta.py [hunk 8]
```diff
@@ -106,6 +106,29 @@ def endModelling(self):
...
+    # function to get arguments of a function
+    def parameters(self, func):
+        def wrapper(*args, **kwargs):
+            # to get parameters of function that are passed
+            functionParameters = inspect.signature(func).bind(*args, **kwargs).arguments
+            functionParameters = dict(functionParameters)
+            # to get values that are not passed and defautlt values
+            defaultParameters = inspect.signature(func)
+            for param in defaultParameters.parameters.values():
+                if ( param.default is not param.empty):
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:120]: i think you can write the nested if stmt in one line

**Review by Priyanka-Vevesta — CHANGES_REQUESTED** [review]

File: vevestaX/vevesta.py (hunk 8 above)
Comment (sparsh459) [reply vevestaX/vevesta.py]: done

**Review by Priyanka-Vevesta — APPROVED** [review]

**Pull request merged**